### PR TITLE
Websocket and http server for GUI server

### PR DESCRIPTION
**Pull request by protobird-git**
GUI server based on websocket.
    
1) Provides a class of websocket and http server used for GUI
2) Based on websockets module which needs python 3.4 or beyond.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 language: python
 sudo: required
 python:
-  - "2.7"
+  - "3.4"
```
Comment (mbjorge) [inline .travis.yml]: Maybe also add a note in the disclaimer that we only test/support python3

File: README.md [hunk 2]
```diff
@@ -36,6 +37,7 @@ $ apt-get install libgraphviz-dev
 $ pip install pygraphviz \
      --install-option="--include-path=/usr/include/graphviz" \
      --install-option="--library-path=/usr/lib/graphviz/"
+$ pip install websockets
```
Comment (mbjorge) [inline README.md]: Does it make sense to add a section in the readme about how to start up / use the GUI?

File: gui/websocket_server.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    self._data_callback = None
+    self._error_callback = None
+
+  @asyncio.coroutine
+  def _OnConnected(self, websocket, path):
+    logging.debug('Websocket is connected: %s, path=%s', str(websocket), path)
+    try:
+      while True:
+        data = yield from websocket.recv()
+        logging.log(1, 'Got a data from a websocket: %s', str(websocket))
```
Comment (mbjorge) [inline gui/websocket_server.py]: I think logging.debug is more clear; the "1" is a bit mysterious (here and below)

File: gui/web_server.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+class WebServer(object):
+  """An abstract class of web server to interact with GUI clients.
+
+  Underlying implementation can be based on http or websocket, or support
+  SSL or not.
+  """
+
+  @abc.abstractmethod
+  def Start(self, port, data_callback, error_callback, websocket_url,
```
Comment (mbjorge) [inline gui/web_server.py]: the websocket_url, index_url, html_url, and html_dir seem maybe a bit specific to the WebsocketServer below for an abstract interface

File: README.md (hunk 2 above)
Comment (protobird-git) [reply README.md]: GUI itself is not usable yet. Let me do that when GUI is integrated in test_driver.py

File: gui/websocket_server.py (hunk 3 above)
Comment (protobird-git) [reply gui/websocket_server.py]: logging.DEBUG is 10. This log messages are less important than logging.DEBUG.

File: gui/web_server.py (hunk 4 above)
Comment (protobird-git) [reply gui/web_server.py]: I defined WebServer considering we may choose different implementation of websocket or web server, for example, to support python 2 though current websocket needs python 3. Those argument are not websocket specific. They would be valid for other websocket/webserver implementations.

File: .travis.yml (hunk 1 above)
Comment (protobird-git) [reply .travis.yml]: Done in README.md

**Comment by protobird-git** [comment]
We used 1, 2 and 3 for more detailed debug messages. At some point, better to define/use logging.fine(), finer(), finest().

**Pull request merged**